apps/docs/pages/guides/getting-started/quickstarts/kotlin.mdx

```diff
@@ -0,0 +1,193 @@
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={4}>
+
+    <StepHikeCompact.Details title="Install the serializable plugin">
+    Open `build.gradle` (app), add serialization plugin to use annotation for data parsing. Please note that the plugin version should be the same with Kotlin version in your app.
+    <a href="https://github.com/JetBrains/kotlin" style={{ marginRight: '8px' }}> 
+          <img src="https://img.shields.io/maven-central/v/org.jetbrains.kotlin/kotlin-maven-plugin.svg?label=kotlin_version" alt="latest stable supabase-kt version"/> 
+    </a>
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+      ```kotlin
+        plugins {
+            ...
+            id 'org.jetbrains.kotlin.plugin.serialization' version '$kotlin_version'
+            ...
+        }
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={5}>
+
+    <StepHikeCompact.Details title="Initialize the Supabase client">
+    You can create a whenever you need to perform some API call.
+    However, it is recommended that we should provide this kind of instance by using dependency injection library like Hilt. So see the code below to provide them properly. This is better for testability and maintainability
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+      ```kotlin
+      // Handy create instance when need
+      val client = createSupabaseClient(
+          supabaseUrl = "https://xyzcompany.supabase.co", 
+          supabaseKey = "public-anon-key"
+        ) {
+              install(GoTrue)
+              install(Postgrest)
+              install(Storage)
+          }
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={6}>
+
+    <StepHikeCompact.Details title="Create data transfer object">
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+      ```kotlin
+      @Serializable
+      data class ProductDto(
+          @SerialName("productid")
+          val productId: String,
+          @SerialName("name")
+          val name: String,
+          @SerialName("description")
+          val description: String,
+          @SerialName("price")
+          val price: Double,
+          @SerialName("image")
+          val image: String,
+          @SerialName("category")
+          val category: String,
+          @SerialName("nutrition")
+          val nutrition: String,
+          @SerialName("_id")
+          val _id: Int,
+      )
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={7}>
+
+    <StepHikeCompact.Details title="Query data from the app">
+      
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+      ```kotlin
+      suspend fun getProducts() {
+        val result = client.postgrest["products"]
+            .select().decodeList<ProductDto>()
+        // Handle result data for next step
+      }
+      ```
+    </StepHikeCompact.Code>
```

Comment:
Maybe it would be good to add another step showing the full activity or something like that? Without, this function seems a bit "alone", as there is nothing really going on (see flutter guide)

Reply:
You are right. I made it "less alone" 😂. Things should be provided from when we get the data from layer to layer until they are displayed to the view.